pkg/services/ngalert/ngalert.go

```diff
@@ -717,6 +732,40 @@ func configureHistorianBackend(
 	return nil, fmt.Errorf("unrecognized state history backend: %s", backend)
 }
 
+func configureNotificationHistorian(
+	ctx context.Context,
+	featureToggles featuremgmt.FeatureToggles,
+	cfg setting.UnifiedAlertingNotificationHistorySettings,
+	met *metrics.NotificationHistorian,
+	l log.Logger,
+	tracer tracing.Tracer,
+) (nfstatus.NotificationHistorian, error) {
+	if !featureToggles.IsEnabled(ctx, featuremgmt.FlagAlertingNotificationHistory) {
+		return nil, nil
+	}
```

Comment:
I would move this one level out and avoid passing `featureToggles` in, would look a bit cleaner.